pkgs/tools/inputmethods/fcitx/unwrapped.nix

```diff
@@ -2,29 +2,69 @@
 , libxml2, enchant, isocodes, icu, libpthreadstubs
 , pango, cairo, libxkbfile, libXau, libXdmcp, libxkbcommon
 , dbus, gtk2, gtk3, qt4, extra-cmake-modules
+, xkeyboard_config, pcre, libuuid, curl, cacert
+, fetchFromGitHub
 }:
 
+let
+  # releases at http://download.fcitx-im.org/fcitx/${name}_dict.tar.xz
+  # contains all data but if we want to compile from source, we need to
+  # fetch them ourselves
+  dicts = let  SPELL_EN_DICT_VER="20121020"; in fetchurl {
+      url = "http://download.fcitx-im.org/data/en_dict-${SPELL_EN_DICT_VER}.tar.gz";
+      sha256 = "1svcb97sq7nrywp5f2ws57cqvlic8j6p811d9ngflplj8xw5sjn4";
+  };
+  # py_stroke= let PY_STROKE_VER="20121124"; in fetchurl {
+  #     url = "http://download.fcitx-im.org/data/py_stroke-${PY_STROKE_VER}.tar.gz";
+  #     sha256 = "0j72ckmza5d671n2zg0psg7z9iils4gyxz7jgkk54fd4pyljiccf";
+  # };
+  # py_table= let PY_TABLE_VER="20121124"; in fetchurl {
+  #     url = "http://download.fcitx-im.org/data/py_table-${PY_TABLE_VER}.tar.gz";
+  #     sha256 = "011cg7wssssm6hm564cwkrrnck2zj5rxi7p9z5akvhg6gp4nl522";
+  # };
+  table = fetchurl {
+      url = "http://download.fcitx-im.org/data/table.tar.gz";
+      sha256 = "1dw7mgbaidv3vqy0sh8dbfv8631d2zwv5mlb7npf69a1f8y0b5k1";
+  };
+  # pinyin_data = fetchurl {
+  #     url = "http://download.fcitx-im.org/data/pinyin.tar.gz";
+  #     sha256 = "1qfq5dy4czvd1lvdnxzyaiir9x8b1m46jjny11y0i33m9ar2jf2q";
+  # };
+in
 stdenv.mkDerivation rec {
   name = "fcitx-${version}";
-  version = "4.2.9.1";
+  version = "4.2.9.5";
 
-  src = fetchurl {
-    url    = "http://download.fcitx-im.org/fcitx/${name}_dict.tar.xz";
-    sha256 = "0xvcmm4yi7kagf55d0yl3ql5ssbkm9410fwbz3kd988pchichdsk";
+  src = fetchFromGitHub {
+    owner = "fcitx";
+    repo = "fcitx";
+    rev = "0ee405a59795232c08cf0cc0b1bf20454fc4cb28";
```

Comment:
Nit: Prefer stable versions with `patches` when possible, else rename the package to [unstable](https://nixos.org/nixpkgs/manual/#sec-package-naming).
Also the hash seem to have changed:

```
these derivations will be built:
  /nix/store/imslrw5bmqb2xmjg4pxj523vn1hsradq-source.drv
  /nix/store/8kbjdvj2nxpjb5n8isp9j2bqfb3qh3q6-fcitx-4.2.9.5.drv
building path(s) ‘/nix/store/a0cvbp1vpy2yihw29v9d0bjm1q0wj316-source’

trying https://github.com/fcitx/fcitx/archive/0ee405a59795232c08cf0cc0b1bf20454fc4cb28.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   153    0   153    0     0    153      0 --:--:-- --:--:-- --:--:--   171
100 2604k  100 2604k    0     0   520k      0  0:00:05  0:00:05 --:--:--  590k
unpacking source archive /tmp/nix-build-source.drv-0/0ee405a59795232c08cf0cc0b1bf20454fc4cb28.tar.gz
output path ‘/nix/store/a0cvbp1vpy2yihw29v9d0bjm1q0wj316-source’ has r:sha256 hash ‘0rv69bacdvblka85dakz4ldpznrgwj59nqcccp5mkkn1rab4zh1r’ when ‘1j5wqj1zcihf171p3zc8g6sn4xy5jpcxg3wmiqn32cc6226n19kb’ was expected
cannot build derivation ‘/nix/store/8kbjdvj2nxpjb5n8isp9j2bqfb3qh3q6-fcitx-4.2.9.5.drv’: 1 dependencies couldn't be built
error: build of ‘/nix/store/8kbjdvj2nxpjb5n8isp9j2bqfb3qh3q6-fcitx-4.2.9.5.drv’ failed
```